src/main/java/hudson/plugins/warnings/parser/DrMemoryParser.java

```diff
@@ -0,0 +1,221 @@
+ * @author Wade Penson
+ */
+@Extension
+public class DrMemoryParser extends RegexpDocumentParser {
+    private static final long serialVersionUID = 7195239138601238590L;
+
+    /**
+     * Regex pattern for Dr. Memory errors and warnings.
+     *
+     * The pattern first tries to capture the header of the error message
+     * with the stack trace. If there happens to be no stack trace for some
+     * reason, only the header with any optional notes will be captured.
+     * This is reflected by the ( | ).
+     *
+     * The body can consist of a stack trace and a notes section. Both the
+     * stack trace and notes can consist of multiple lines. A line in the stack
+     * trace starts with "#" and a proceeding number and the lines in the notes
+     * section start with "Note: ". The first part of pattern will match the
+     * the lines that start with "#" until it can't find a line that starts with
+     * "#". If the next line starts with "Note: ", it will match the rest of the
+     * lines until there are two consecutive newlines (which indicates that the
+     * end of the error has been reached).
+     *
+     * Note: Groups can have trailing whitespace.
+     */
+    private static final String DR_MEMORY_WARNING_PATTERN =
+        "(?:Error #\\d+: ([\\s\\S]+?)\\r?\\n(# \\d+ [\\s\\S]*?\\r?\\n)(?=[^#])(Note: [\\s\\S]*?\\r?\\n\\r?\\n)?|" +
+        "Error #\\d+: ([\\s\\S]+?)\\r?\\n\\r?\\n)";
+
+    /** The index of the regexp group capturing the header of the error or warning. */
+    private static final int HEADER_GROUP = 1;
+
+    /** The index of the regexp group capturing the stack trace of the error or warning. */
+    private static final int STACK_TRACE_GROUP = 2;
+
+    /** The index of the regexp group capturing the notes of the error or warning. */
+    private static final int NOTES_GROUP = 3;
+
+    /** Regex pattern to extract the file path from a line. */
+    private static final Pattern FILE_PATH_PATTERN =
+        Pattern.compile("#\\s*\\d+.*?\\[(.*\\/?.*):(\\d+)\\]");
+
+    /** The index of the regexp group capturing the file path of a location in the stack trace. */
+    private static final int FILE_PATH_GROUP = 1;
+
+    /** The index of the regexp group capturing the line number of a location in the stack trace. */
+    private static final int LINE_NUMBER_GROUP = 2;
+
+    /** Regex pattern to extract the jenkins path from file path. */
+    private static final Pattern JENKINS_PATH_PATTERN =
+        Pattern.compile(".*?(\\/jobs\\/.*?\\/workspace\\/|workspace\\/)");
+
+    /**
+     * Creates a new instance of {@link DrMemoryParser}.
+     */
+    public DrMemoryParser() {
+        super(Messages._Warnings_DrMemory_ParserName(),
+                Messages._Warnings_DrMemory_LinkName(),
+                Messages._Warnings_DrMemory_TrendName(),
+                DR_MEMORY_WARNING_PATTERN, false);
+    }
+
+    @Override
+    protected Warning createWarning(final Matcher matcher) {
+        String message = "";
```

Comment:
Normally, concatenation of strings is better done using a StringBuilder. But I think the VM is already optimizing this.